Add to functional requirements

lafrenierejm opened the pull request:
The only functional requirements that I know are still missing are ones pertaining to
* weather protection
* scheduling of kitchen appliances
* detectors for smoke, natural gas

lafrenierejm committed:
Elaborate on FRs for order of applying preferences

lafrenierejm committed:
Add FRs pertaining to external systems

lafrenierejm committed:
Add FRs pertaining to monitoring internal systems

lafrenierejm committed:
Add FR pertaining to manual overrides

lafrenierejm committed:
Add FR pertaining to notifying users

lafrenierejm committed:
Add FRs pertaining to smoke detection

lafrenierejm committed:
Add FR that users be notified of severe weather

lafrenierejm committed:
Add FRs pertaining to scheduling automation

jmitash commented on team13phase1.tex:
```diff
@@ -56,6 +56,24 @@
+      The system shall be able to determine when the refrigerator is empty.
+    \end{fr}
+  \item\label{fr:internal:smoke_detector}
+    The system shall monitor properties' smoke detectors.
```
FRs like these where something is being monitored, watched, or polled need to have an indicator and an reaction to that indicators. For example, `The system shall monitor each property's smoke detectors and send a high priority alert to the property manager when smoke is detected.` You might want to combine some of the FRs from the WLAN section to here, so that these can have more meaning.

lafrenierejm replied:
But the monitoring and the notifying are two discrete functionalities, are they not?  Certainly the latter depends on the former, but I think that should be expressed by referencing the former not by combining both pieces of functionality into a single FR.

jmitash commented on team13phase1.tex:
```diff
@@ -67,23 +85,47 @@
+  \item
+    The system shall allow users to schedule the application of profiles from \ref{fr:customizable:multiple_profiles}.
+  \item
+    Every aspect of the system that has automation shall have physical, manual overrides.
```
The system shall provide physical overrides for automated aspects of the system.

jmitash commented on team13phase1.tex:
```diff
@@ -67,23 +85,47 @@
+  \item\label{fr:software:ui}
+    The system shall present a graphical user interface (GUI) to its end users.
+  \item\label{fr:software:devices}
+    The system's GUI shall support the following classes of devices:
```
I feel like this should be under the UI section. Perhaps move the list of devices to a broader scope and just reference them from the UI section?

jmitash commented on team13phase1.tex:
```diff
@@ -94,22 +136,24 @@
+    The system shall allow a property's guests to remotely view the contents of that property's refrigerator.
   \end{fr}
 \item
   The system shall allow users to set personal preferences for property configuration.
```
Maybe this (line 143 and below) should be reorganized to something like this:
```
The system shall allow users to set personal preferences for property configuration.
    The system shall provide a preferences control panel accessible from the devices listed in \ref{fr:software:devices}.
        The control panel shall include preferences for the aspects listed in \ref{fr:customizable}.
    The system shall persist the user preferences and maintain them for any subsequent properties that have not already had preference profiles manually set.
    The system shall allow property managers to determine which aspects of the property their other users may set preferences for.
    The system shall apply the user preferences of the user with the highest role when there are conflicting preferences.
        [Roles]
        The system shall use the average of the highest ranking users when multiple users of the highest role have conflicting preferences.
```

jmitash commented on team13phase1.tex:
```diff
@@ -149,7 +195,21 @@
+  \item
+    A property's owner, managers, guests, and maintenance staff shall be notified if the property's smoke detector senses any smoke.
+  \item
+    If data collected in \ref{fr:external:weather} indicates ongoing or imminent severe weather at a property, the owner, managers, and guests of that property are notified of that sever weather.
```
When describing FRs, we can really only describe the behavior of the system. It becomes problematic to say "The property manager will be notified" because the system doesn't control the property manager. The best we can do is say something like "The system shall send a notification/alert to the property manager's phone when [event] occurs"

lafrenierejm committed:
Merge pull request #3 from lafrenierejm/add-project-description

Add project description

lafrenierejm committed:
Move FR for device access under FR for UI

lafrenierejm committed:
Revise FR pertaining to user preferences

This commit incorporates feedback offered by knoxcorner in his review
of PR #1.
